en/ros2/px4_sdk.md

```diff
@@ -0,0 +1,336 @@
+2. Create an instance of [px4_sdk::OffboardActuatorControls](https://github.com/PX4/px4_sdk/blob/main/px4_sdk_cpp/include/px4_sdk/control/offboard_actuators.hpp) in the constructor of your mode
+3. Call the `set` method to control the actuator(s). This can be done independently of any active setpoints.
+
+### Telemetry
+
+Telemetry, such as local or global position estimates can be found under [px4_sdk/odometry](https://github.com/PX4/px4_sdk/tree/main/px4_sdk_cpp/include/px4_sdk/odometry).
+
+### Failsafes and Mode Requirements
+
+Each mode has a set of requirement flags. These are generally automatically set, depending on which objects are used within the context of a mode. For example when adding manual control input with
+
+```cpp
+_manual_control_input = std::make_shared<px4_sdk::ManualControlInput>(*this);
+```
+
+the requirement flag for manual control gets set.
+Specifically, setting a flag has the following consequences in PX4, if the condition is not met:
+
+- arming is not allowed, while the mode is selected
+- when already armed, the mode cannot be selected
+- when armed and the mode is selected, the relevant failsafe is triggered (e.g. RC loss for the manual control requirement).
+  Check the [safety page](../config/safety.md) for how to configure failsafe behavior.
+  A failsafe is also triggered when the mode crashes or becomes unresponsive while it is selected.
+
+It is possible to manually update any mode requirement after the mode is registered. For example to add home position as requirement:
+
+```cpp
+modeRequirements().home_position = true;
+```
+
+#### Deferring Failsafes
+
+A mode or mode executor can temporarily defer non-essential failsafes.
+To do so, use the method `deferFailsafesSync`. And to get notified when a failsafe would be triggered, override the method `void onFailsafeDeferred()`.
+Check the [integration test](https://github.com/PX4/px4_sdk/blob/main/px4_sdk_cpp/test/integration/overrides.cpp#L167) for an example.
+
+### Assigning a Mode to an RC Switch or Joystick Action
+
+External modes can be assigned to RC switches or joystick actions.
+When assigning a mode to an RC switch, you need to know the index.
+Use `commander status` while the mode is running to get that information.
+For example:
+
+```plain
+   INFO  [commander] External Mode 1: nav_state: 23, name: My Manual Mode
+```
+
+means you would select **External Mode 1** in QGC:
+![QGC Mode Assignment](../../assets/middleware/ros2/px4_sdk/qgc_mode_assignment.png)
+
+:::note
+PX4 ensures a given mode is always assignment to the same index by storing a hash of the mode name.
+This makes it independent of startup ordering in case of multiple external modes.
+:::
+
+### Replacing an Internal Mode
+
+An external mode can replace an existing internal mode, such as RTL. By doing so, whenever RTL gets selected (through the user or a failsafe situation), the external mode is used instead of the internal one.
```

Comment:
```suggestion
An external mode can replace an existing internal mode, such as [Return](../flight_modes/return.md) mode (RTL).
By doing so, whenever RTL gets selected (through the user or a failsafe situation), the external mode is used instead of the internal one.
```